src/main/java/com/amihaiemil/versioneye/MkVersionEye.java

```diff
@@ -88,7 +106,29 @@ public VersionEye trusted() throws IOException {
 
     @Override
     public Me me() {
+        if(this.authenticated == null) {
+            throw new IllegalStateException("No Authenticated User.");
+        }
         return new MkMe(this.server, this.authenticated);
     }
+    
+    /**
+     * Add authenticated user to the MkServer.
+     */
+    private void addAuthenticatedUserToServer() {
+        JsonArray array = this.server.storage().build()
+            .getJsonArray("authenticated");
+        if(array == null) {
```

Comment:
@SherifWaly I see... ok, then leave it like this. I'll think of how we can get rid of it in some other way :D